functions/lib/crowd_webhook.js

```diff
@@ -0,0 +1,87 @@
+module.exports = CrowdWebhook
+
+function hasProperty (object, property) {
+  return Object.prototype.hasOwnProperty.call(object, property)
+}
+
+function CrowdWebhook (groupMappings, github) {
+  if (!groupMappings) {
+    // Use default mapping if one is not provided
+    groupMappings = require('./crowd_webhook_config')
+  }
+
+  async function processMembership (user, group, filter, add) {
+    const noFilter = !Array.isArray(filter) || !filter.length
+    const mappings = groupMappings[group]
+    for (const i in mappings) {
+      const mapping = mappings[i]
+      if (hasProperty(mapping, 'githubOrg') && // handle Github membership
+              (noFilter || filter.includes('github')) &&
+              hasProperty(user, 'githubId')) {
+        try {
+          if (add) {
+            console.log(`Adding ${user.githubId} to ${mapping.githubOrg}:${mapping.team}`)
+            // TODO: save some API calls by not always calling this
+            await github.createTeam(mapping.githubOrg, mapping.team)
+            await github.addUser(user.githubId, mapping.githubOrg, mapping.team)
+            // TODO: handle response?
```

Comment:
Is there anything to handle? If I'm not mistaken octokit thrown  an exception in case of errors, which should be sufficient for at least logging the error.